src/Routes.js

```diff
@@ -11,9 +11,15 @@ import AssetModelViewer from "./Components/Admin/AssetModelViewer/assetModelView
 import { UserContext } from "./Providers/UserProviders";
 
 export default function Routes() {
-    const { isAdmin } = useContext(UserContext);
+    const { isLoading, user } = useContext(UserContext);
 
-    return isAdmin ? (
+    useEffect(() => {
+        if (isLoading) {
+            return;
+        }
+    }, [isLoading]);
```

Comment:
What does this do?

Reply:
I'll try removing it and see if it works but basically the idea is that if you are loading the routes without the user set (but in a loading state), recursively "wait" (ie. return nothing) until it is set, and then the rest of the routes file will load.